### PR TITLE
Fixing tooltip component

### PR DESCRIPTION
**Pull request by axeleriksson147**
Tooltip doesn't disappear when hovering when a new DOM element overlays the previous one. Using an updated version of @floating-ui/react I can't reproduce it.

### CODE REVIEW

File: packages/components/src/components/tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -1,176 +1,155 @@
...
-    </div>
-    }
-  </>
-};
-
-Tooltip.defaultProps = {
-  type: 'hint',
-};
-
-Tooltip.propTypes = {
```
Comment (axeleriksson147) [inline packages/components/src/components/tooltip/Tooltip.tsx:162]: @antoinerollindev @aurelie-forzani @benoit-sy do you get usage from prop-styles package? I feel like TypeScript should be enough and it's a burden having both maintained.

Comment (benoit-sy) [reply packages/components/src/components/tooltip/Tooltip.tsx:162]: @axeleriksson147 The only use-case where PropTypes still makes sense over pure TypeScript is for when dynamic data are passed to components.
So we could maybe plan for a global refactor PR where we ditch entirely prop-types in favour or TS types/interfaces (that most of the time we already have)

Comment (axeleriksson147) [reply packages/components/src/components/tooltip/Tooltip.tsx:162]: Yeah, and in that case it's about getting an "easier" error message but the component will error either way I believe.

**Review by benoit-sy — APPROVED** [review]
✅ for a `v4.2.1`

**Pull request merged**